xbmc/cores/VideoPlayer/DVDMessage.h

```diff
@@ -271,6 +266,28 @@ class CDVDMsgPlayerSetSpeed : public CDVDMsg
 
 };
 
+class CDVDMsgOpenFile : public CDVDMsg
+{
+public:
+  struct FileParams
+  {
+    CFileItem m_item;
+    CPlayerOptions m_options;
+  };
+
+  CDVDMsgOpenFile(FileParams params)
+  : CDVDMsg(PLAYER_OPENFILE)
+  , m_params(params)
```

Comment:
`m_params(std::move(params))` assuming that both params have move constructors. If not, then passing by reference `(const FileParams &params)` is the next best thing